### PR TITLE
.sdf File Reader

### PR DESCRIPTION
**Pull request by joegomes**
This PR adds:
- A new input file type, .sdf.  The structures are read directly to RDKit `Mol` objects that are used for featurization.  This allows one to use optimized geometries rather than ones auto-generated from SMILES strings for structure-sensitive featurization techniques such as the `CoulombMatrix`.
- A new notebook, `examples/quantum_machine_gdb1k`.  This notebook contains a demonstration of how to correctly import .sdf files into DeepChem and build simple random forest and kernal ridge regression models.

Notes:
-When reading geometries from xxx.sdf file, tasks should be stored in a xxx.sdf.csv file.

### CODE REVIEW

File: deepchem/featurizers/__init__.py [hunk 1]
```diff
@@ -142,7 +142,9 @@ def featurize(self, mols, parallel=False, client_kwargs=None,
         if verbosity is not None and i % log_every_n == 0:
           log("Featurizing %d / %d" % (i, len(mols)))
         if mol is not None:
-          features.append(self._featurize(mol))
+          myfeatures = self._featurize(mol)
+          features.append(myfeatures)
+          #features.append(self._featurize(mol))
```
Comment (rbharath) [inline deepchem/featurizers/__init__.py]: Could you remove the comment here?

**Comment by rbharath** [comment]
@joegomes: This looks great! As a request before merging, could you put together some simple tests for the SDF reader? I think the file-loading module is getting complicated, so I'm thinking we will need to refactor soon, so it would be great to have tests that specify what correct behavior looks like.

**Comment by joegomes** [comment]
@rbharath I've added a test similar to the ones found in `featurizers/tests/test_featurized_samples.py`.  The test reads 10 water geometries, does a CoulombMatrixEig featurize/split, and then checks for the correct number of samples in train, test, and valid sets.  Is this what you had in mind?

**Comment by joegomes** [comment]
@rbharath I've updated the notebook models, they are now giving OK performance given the dataset size (GDB 1K subset).  While I was comparing results with coulomb matrices given in the original QuantumMachine dataset, I found a bug in the `CoulombMatrix` class.  The length units of the distance matrix used to create the coulomb matrix should be in bohr (atomic units) rather than angström (RDKit units).  This change gives good agreement with the reference coulomb matrices and improves performance.

**Comment by rbharath** [comment]
@joegomes Great! Will go ahead and merge. Thanks for putting this together :-)

**Pull request merged**